internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -343,3 +355,95 @@ func (k *KafkaManager) updateClusterStatusCapacityMaxMetric(c services.KafkaStre
+				kafka.ID, kafka.InstanceType, kafka.SizeId,
+			))
+			continue
+		}
+
+		expiresAt := kafka.ExpiresAt
+		// If lifespan seconds is not defined, expiration is determined based on the organisation's active subscription
+		if instanceSize.LifespanSeconds == nil {
+			orgBillingModelID := fmt.Sprintf("%s-%s", kafka.OrganisationId, kafka.ActualKafkaBillingModel)
+			active, ok := subscriptionStatusByOrgAndBillingModel[orgBillingModelID]
+			if !ok {
+				isActive, err := k.kafkaService.IsSubscriptionActive(kafka)
+				if err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to get subscription status of kafka instance %q", kafka.ID))
+					continue
+				}
+				subscriptionStatusByOrgAndBillingModel[orgBillingModelID] = *isActive
+				active = *isActive
+			}
+
+			if active {
+				// if subscription is active, set expires_at to null
+				if err := expiresAt.Scan(nil); err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to set expires_at value for kafka instance %q to NULL", kafka.ID))
+					continue
+				}
+			} else {
+				// if subscription is not valid and expires_at is not already set, set its value based on the current time and grace period allowance
+				if !kafka.ExpiresAt.Valid {
+					// set expires_at to now + grace period days
+					billingModel, err := k.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+					if err != nil {
+						svcErrors = append(svcErrors, errors.Wrapf(err, "failed to get billing model of kafka instance %q", kafka.ID))
+						continue
+					}
+
+					duration, err := time.ParseDuration(fmt.Sprintf("%dh", billingModel.GracePeriodDays*24))
+					if err != nil {
+						svcErrors = append(svcErrors, errors.Wrapf(err, "failed to parse expiration duration for kafka instance %q", kafka.ID))
+						continue
+					}
+
+					if err := expiresAt.Scan(time.Now().Add(duration)); err != nil {
+						svcErrors = append(svcErrors, errors.Wrapf(err, "failed to set expires_at value for kafka instance %q", kafka.ID))
+						continue
+					}
+				}
+			}
+		} else {
+			// any Kafka instance types with lifespan seconds defined will have their expires_at value set on creation.
+			// this is only temporary to ensure that the expires_at value is set for any Kafka instances created during the rollout of this change.
+			if !kafka.ExpiresAt.Valid {
+				// set expires_at to created_at + lifespanSeconds
+				duration, err := time.ParseDuration(fmt.Sprintf("%ds", *instanceSize.LifespanSeconds))
+				if err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to parse expiration duration for kafka instance %q", kafka.ID))
+					continue
+				}
+
+				if err := expiresAt.Scan(kafka.CreatedAt.Add(time.Duration(duration))); err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to set expires_at value for kafka instance %q", kafka.ID))
+					continue
+				}
+			}
+		}
+
+		if expiresAt != kafka.ExpiresAt {
```

Comment:
This is doing struct comparison. Are we aware of this?

Reply:
This was another think I though as well. But with the change I suggested, we don't need this anymore.